rasa/src/Rasa/Internal/Directive.hs

```diff
@@ -127,7 +127,10 @@ replaceRange r txt = range r .= txt
 
 -- | Inserts text into the buffer at the given Coord.
 insertAt :: Coord -> Y.YiString -> BufAction ()
-insertAt c txt = range (Range c c) .= txt
+insertAt c txt = do
+  let r = Range c c
+  liftAction $ dispatchEvent (BufTextChanged r)
+  range r .= txt
 
 -- | Runs the given function over the text in the range, replacing it with the results.
 overRange :: Range -> (Y.YiString -> Y.YiString) -> BufAction ()
```

Comment:
If we add the new text value into the event (as I suggest below) then the dispatch looks like this: (using [`<%=`](https://hackage.haskell.org/package/lens-4.15.1/docs/Control-Lens-Lens.html#v:-60--37--61-) from the lens library)

```haskell
overRange r f = do
  newText <- range r <%= f
  liftAction $ dispatchEvent (BufTextChanged r newText)
```

What do you think about something like that?